autorel/dist_tarball_builder/dist_tarball.py

```diff
@@ -0,0 +1,158 @@
+"""
+    module : dist_tarball
+    This module is responsible for building syslog-ng from source
+    in a docker environment and providing functionality of exporting
+    the generated distribution tarball.
+"""
+import logging
+import tempfile
+import pygit2
+import os
+import sys
+import glob
+from docker_cli import Docker
+
+
+
+class DistTarballBuilder(object):
+    """
+        Responsible for downloading souce code and carry out
+        build process to generate the distribution tarball.
+    """
+    def __init__(self, project, source_url, image_name):
+        self._project = project
+        self._source_url = source_url
+        self._image_name = image_name
+        self._docker_cli = Docker()
```

Comment:
Yeah, you are suggesting passing `Docker` class an instance of `DistTarballBuilder` and then running the required methods ?